legend-engine-xts-analytics/legend-engine-xts-analytics-quality/legend-engine-xt-analytics-quality-api/src/main/java/org/finos/legend/engine/api/analytics/DataspaceQualityAnalytics.java

```diff
@@ -0,0 +1,242 @@
+import org.finos.legend.pure.generated.core_analytics_quality_propertyChecks;
+import org.finos.legend.pure.m3.coreinstance.meta.pure.metamodel.function.ConcreteFunctionDefinition;
+import org.finos.legend.pure.m3.coreinstance.meta.pure.metamodel.function.property.Property;
+import org.finos.legend.pure.m3.coreinstance.meta.pure.metamodel.type.Class;
+import org.finos.legend.pure.m3.coreinstance.meta.pure.metamodel.type.Enum;
+import org.finos.legend.pure.m3.coreinstance.meta.pure.metamodel.type.Enumeration;
+import org.finos.legend.pure.runtime.java.compiled.execution.CompiledExecutionSupport;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Api(tags = "Analytics - Quality")
+@Path("pure/v1/analytics/quality")
+public class DataspaceQualityAnalytics
+{
+    private final ModelManager modelManager;
+    private List<String> messages;
+
+    public DataspaceQualityAnalytics(ModelManager modelManager)
+    {
+        this.modelManager = modelManager;
+    }
+
+    @POST
+    @Path("check")
+    @ApiOperation("Checks the quality of provided Data space")
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response checkDataSpaceConstraints(PureModelContextData pureModelContextData)
+    {
+        messages = new ArrayList<>();
+        PureModel pureModel = this.modelManager.loadModel(pureModelContextData, null, null, null);
+
+        List<PackageableElement> classes = pureModelContextData.getElements().stream().filter(e -> e instanceof org.finos.legend.engine.protocol.pure.v1.model.packageableElement.domain.Class).collect(Collectors.toList());
+        List<PackageableElement> associations = pureModelContextData.getElements().stream().filter(e -> e instanceof org.finos.legend.engine.protocol.pure.v1.model.packageableElement.domain.Association).collect(Collectors.toList());
+        List<PackageableElement> enumerations = pureModelContextData.getElements().stream().filter(e -> e instanceof org.finos.legend.engine.protocol.pure.v1.model.packageableElement.domain.Enumeration).collect(Collectors.toList());
+        List<PackageableElement> functions = pureModelContextData.getElements().stream().filter(e -> e instanceof org.finos.legend.engine.protocol.pure.v1.model.packageableElement.domain.Function).collect(Collectors.toList());
+
+        CompiledExecutionSupport es = pureModel.getExecutionSupport();
+
+        if (classChecks(classes, pureModel, es) && associationChecks(associations, pureModel, es) && enumerationChecks(enumerations, pureModel, es) && functionChecks(functions, pureModel, es))
+        {
+            return Response.ok().build();
+        }
+        else
+        {
+            return Response.status(400).entity("Dataspace does not match the constraints - " + messages).build();
```

Comment:
it would be good to have a modelled response here instead of an arbitrary String. String is unclear from a client perspective, and really does not allow any useful processing from client applications.
maybe something like
{
    errors: []
}
where error has a type (class, enum, assocation etc) and a message. In Pure there is also a concept of a quality check group to be able to classify the issue type